tools/packaging/common/envoy_bootstrap.json

```diff
@@ -231,7 +231,7 @@
               "endpoint": {
                 "address":{
                   "pipe": {
-                    "path": "{{ .config.ConfigPath }}/SDS"
+                    "path": "./var/run/secrets/workload-spiffe-uds/socket"
```

Comment:
My concern with using the same path for both is for workload restarts. I am pretty sure this can happen in Kubernetes since the socket is now in a volume mount, and it definitely can with VMs.

1. Run 1 - we detect no socket, so we create it and start server.
2. restart for various reasons
3. Run 2 - we detect a socket, so we connect to it and skip our server. Socket never sends anything

Do we prevent this somehow?

Reply:
In `3.`, do you mean that a socket injected by SPIRE never sends anything ?

Reply:
In this flow I am talking only standard Istio. So no, 3 is the socket created by the first istio-agent run. The socket file is not removed when the process restarts

Reply:
Got it, we are not handling that scenario. I think that scenario could be addressed by doing a test call to the socket from the istio agent using a SDS client, to check that the socket is responding. Not sure what would the best approach if the socket is not responding, though.

Reply:
Should we make it as another path but not use the same path?  Imagine we should have other cases that might need use another sds address.

Reply:
@howardjohn, I've looking into the problem with an old socket hanging around and I think we could check if it's responsive and use it, otherwise try to delete it and start the default SDS server:

```
socketExists := socketFileExists(security.WorkloadIdentitySocketPath)

// if the socket exists, check whether it's responsive:
// - in the case it's responsive: istio-agent doesn't start its own SDS server
// - in the case it's not responsive, it's probably an old socket from a previous istio-agent instance
//   and then try to remove the socket and set the flag "socketExist" to false
//   if the socket cannot be removed, its another problem and the startup should fail
if socketExists {
	conn, err := net.Dial("unix", security.WorkloadIdentitySocketPath)
	if err != nil {
		err = os.Remove(security.WorkloadIdentitySocketPath)
		if err != nil {
			return nil, fmt.Errorf("error connecting to existing socket and could not be removed: %v", err)
		}
		// setting socket existence flag to false
		socketExists = false
	} else {
		conn.Close()
		log.Info("SDS socket found. Istio SDS Server won't be started")
	}
}

if !socketExists { 
          ///  ...
 }
```

What do you think?

Reply:
Hmm, I am surprised that works. What is the socket returning? I would expect a GRPC server to not send any bytes until we send it some?

Reply:
The socket is returning the settings frame on the first `Read()`, that's why it works. 

But I explored and tested another way that is more robust, using the `grpcDialContext` with a Deadline:

```
ctx, cancel := context.WithDeadline(ctx, time.Now().Add(time.Second))
defer cancel()

conn, err := grpc.DialContext(ctx, fmt.Sprintf("unix:%s", security.WorkloadIdentitySocketPath),
                        grpc.WithTransportCredentials(insecure.NewCredentials()),
		        grpc.FailOnNonTempDialError(true),
			grpc.WithReturnConnectionError(),
			grpc.WithBlock(),
		)
```

Reply:
Makes sense to me. In the happy case it is ~instant right, not 1s?

Reply:
Yes, took around 400 µs in my tests in a kind cluster to check the SPIRE socket.

Reply:
@howardjohn, I added the socket health check and did some refactors on my code to reduce the complexity of the `Run()` method.